### DISCUSSION TITLE
i3d features?

### DISCUSSION THREAD

**Discussion by NitinDatta8** [ekosman/AnomalyDetectionCVPR2018-Pytorch#210] [Announcements]
Hi, 
Does anyone have I3D features for these videos. 
Any link might be helpful.. 

Thank you

**Comment by ekosman** [comment]
@NitinDatta8 
Do you know any decent implementation of this?
I can consider adding this feature to this repository as I'm currently working on fixing some issues here.

**Comment by NitinDatta8** [comment]
Nope @ekosman, I tried a lot but was not successful.
However, I can help you with the colab version for extracting features and training. 
I have them ready... 
If you want I can share it here.

**Comment by ekosman** [comment]
@NitinDatta8 Ok, I'd prefer a pull request but you can also put it here and I'll aggregate everything once done.

**Comment by NitinDatta8** [comment]
Here you go @ekosman 
Feature extraction: https://drive.google.com/file/d/16UGVIqi3idOLUcPB9jNt0liSEqDNNDe3/view?usp=sharing
Training - https://drive.google.com/file/d/1Hiry_yzFYVSIKY6T1puL16YEv4lcOqI0/view?usp=sharing
ROC_AUC - https://colab.research.google.com/drive/1pjfmDTU1gb8KHqWqvO7hFeB9O3eT27i0?usp=sharing


Please let me know if you find anything related to I3D. It will be really helpful..

**Comment by ekosman** [comment]
@NitinDatta8 
Nice, I'll add a notebook directory

**Comment by ezzatmostafa96** [comment]
colab not working

**Comment by NitinDatta8** [comment]
Hi @ezzatmostafa96 , 
I made changes maybe you can check this [LINK](https://github.com/ekosman/AnomalyDetectionCVPR2018-Pytorch/tree/master/notebooks) and run it on colab it will work

**Comment by ezzatmostafa96** [comment]
ok and testing the video for detection?

**Comment by ekosman** [comment]
Hi @NitinDatta8 
Are you familiar with i3d? I'm trying to integrate I3d from `https://github.com/piergiaj/pytorch-i3d` into this repo but have some missing knowledge about the model.

**Comment by santiagosilas** [comment]
I found that Tian et al (2021) made available I3D features in this source code repository for his paper: https://github.com/tianyu0207/RTFM.

I3D Features for UCF Crime Training Set: https://uao365-my.sharepoint.com/personal/a1697106_adelaide_edu_au/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fa1697106%5Fadelaide%5Fedu%5Fau%2FDocuments%2FUCF%5FTrain%5Ften%5Fcrop%5Fi3d%20%281%29


I3D Features for UCF Crime Test Set:  https://uao365-my.sharepoint.com/personal/a1697106_adelaide_edu_au/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fa1697106%5Fadelaide%5Fedu%5Fau%2FDocuments%2FUCF%5FTest%5Ften%5Fcrop%5Fi3d%20%281%29

Did anyone already test these features ?
